lib/test/unit/assertions.rb

```diff
@@ -926,15 +926,39 @@ def assert_nothing_thrown(message=nil, &proc)
       #   assert_in_delta 0.05, (50000.0 / 10**6), 0.00001
       def assert_in_delta(expected_float, actual_float, delta=0.001, message="")
         _wrap_assertion do
-          _assert_in_delta_validate_arguments(expected_float,
-                                              actual_float,
-                                              delta)
-          full_message = _assert_in_delta_message(expected_float,
-                                                  actual_float,
-                                                  delta,
-                                                  message)
+          begin
+            _assert_in_delta_validate_arguments(expected_float,
+                                                actual_float,
+                                                delta,
+                                                type_conversion: false)
```

Comment:
How about just raising here? I think that we should fallback to the `to_f` case for this case because `expected_float`, `actual_float` or `delta` may need to use `to_f`.

```suggestion
            raise unless delta >= 0
```

Or we can call this after `pass = (expected_float - actual_float).abs <= delta` because we can assume that we don't need to `to_f` after the `pass = ...`.